### PR TITLE
Add Backward cpp

### LINKED ISSUE #532: Request: backward-cpp
Name: backward-cpp
Purpose: Pretty printing C++ stacktrace
Repository: https://github.com/bombela/backward-cpp

### PR DESCRIPTION
**Pull request by tmayoff**
This would close #532

### CODE REVIEW

File: subprojects/packagefiles/backward-cpp/meson.build [hunk 1]
```diff
@@ -0,0 +1,16 @@
+project('backward-cpp', 'cpp', version: '1.6')
+
+install_headers('backward.hpp', subdir : 'backward')
+backward = library('backward-cpp', files('backward.cpp'), include_directories: '.', install: true)
```
Comment (jpakkane) [inline subprojects/packagefiles/backward-cpp/meson.build:4]: You don't need `include_directories: '.'`. The current source and build directories are put in the include path by default.

**Comment by jpakkane** [comment]
> File 'subprojects\\backward-cpp-1.6\\backward-cpp.lib' could not be found

This typically means [this](https://mesonbuild.com/FAQ.html#my-project-works-fine-on-linux-and-mingw-but-fails-to-link-with-msvc-due-to-a-missing-lib-file-fatal-error-lnk1181-why).

**Comment by tmayoff** [comment]
Would that require changes to the upstream repo, adding the __declspec(dllexport) stuff? (I'm really not familiar with windows development

**Comment by neheb** [comment]
Yep.

File: subprojects/backward-cpp.wrap [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[wrap-file]
+directory = backward-cpp-1.6
+source_url = https://github.com/bombela/backward-cpp/archive/refs/tags/v1.6.tar.gz
+source_filename = backward-cpp-1.6.tar.gz
+source_hash = c654d0923d43f1cea23d086729673498e4741fb2457e806cfaeaea7b20c97c10
+
+patch_directory = backward-cpp
+
+[provide]
+backward-cpp = backward_dep
```
Comment (neheb) [inline subprojects/backward-cpp.wrap:10]: No newline?

File: subprojects/packagefiles/backward-cpp/meson.build [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+tests = ['test', 'stacktrace', 'rectrace', 'select_signals']
+
+foreach test : tests
+    test('test_' + test, executable('testexe_' + test, 'test/' + test + '.cpp', 'test/_test_main.cpp'))
+endforeach
+
+tests = ['suicide']
+foreach test : tests
+    test('test_' + test, executable('testexe_' + test, 'test/' + test + '.cpp', 'test/_test_main.cpp'))
+endforeach
```
Comment (neheb) [inline subprojects/packagefiles/backward-cpp/meson.build:16]: no newline?

**Comment by ninjaoflight** [comment]
Hi, on a meson port that was done under [SQLiteCpp](https://github.com/SRombauts/SQLiteCpp) as a workaround we the default compilation to static, until the support of a proper macro for dllexport is implemented.

though this is not a [recommended practice](https://mesonbuild.com/Wrap-best-practices-and-tips.html#make-libraries-buildable-both-as-static-and-shared) this can help to get an initial working build on windows

**Comment by xclaesse** [comment]
@ninjaoflight FYI, I proposed a while ago to let meson generate those dll macros: https://github.com/mesonbuild/meson/pull/10199, because everybody gets it wrong. But adding similar header into your project is fine too.

**Comment by tmayoff** [comment]
I forced a static library if on windows for now, if I have some more time I can look into making a PR upstream to add the proper exports

**Comment by neheb** [comment]
```
2023-04-23T01:45:13.1117697Z ld.lld: error: undefined symbol: __declspec(dllimport) SymFunctionTableAccess64
```

**Pull request merged**